code/modules/mob/living/simple_animal/abnormality/teth/tangle.dm

```diff
@@ -0,0 +1,127 @@
+/mob/living/simple_animal/hostile/abnormality/tangle
+	name = "Tangle"
+	desc = "What seems to be a severed head laying in a tangle of hair."
+	icon = 'ModularTegustation/Teguicons/32x32.dmi'
+	icon_state = "tangle"
+	icon_living = "tangle"
+	portrait = "tangle"
+	maxHealth = 1600
+	health = 1600
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1, WHITE_DAMAGE = 0.5, BLACK_DAMAGE = 1, PALE_DAMAGE = 2)
+	melee_damage_lower = 0		//Doesn't attack
+	melee_damage_upper = 0
+	rapid_melee = 2
+	melee_damage_type = WHITE_DAMAGE
+	stat_attack = HARD_CRIT
+	faction = list("hostile")
+	can_breach = TRUE
+	threat_level = TETH_LEVEL
+	start_qliphoth = 2
+	work_chances = list(
+		ABNORMALITY_WORK_INSTINCT = 80,
+		ABNORMALITY_WORK_INSIGHT = list(50, 50, 40, 40, 40),
+		ABNORMALITY_WORK_ATTACHMENT = list(50, 50, 40, 40, 40),
+		ABNORMALITY_WORK_REPRESSION = list(50, 50, 40, 40, 40),
+	)
+	work_damage_amount = 5
+	work_damage_type = WHITE_DAMAGE
+	ego_list = list(
+		/datum/ego_datum/weapon/rapunzel,
+		/datum/ego_datum/armor/rapunzel,
+	)
+//	gift_type =  /datum/ego_gifts/rapunzel
+	abnormality_origin = ABNORMALITY_ORIGIN_WONDERLAB
+
+	var/chosen
+	var/instinct_count
+	var/list/hair_list = list()
+
+/mob/living/simple_animal/hostile/abnormality/tangle/Move()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/tangle/CanAttack(atom/the_target)
+	return FALSE
+
+//Grab a list of all agents and picks one
+/mob/living/simple_animal/hostile/abnormality/tangle/Initialize()
+	. = ..()
+	var/list/potentialmarked = list()
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		if(L.stat >= HARD_CRIT || L.sanity_lost || z != L.z) // Dead or in hard crit, insane, or on a different Z level.
+			continue
+		if(HAS_TRAIT(usr, TRAIT_WORK_FORBIDDEN)) //Don't get non agents
+			continue
+		potentialmarked += L
+	if(potentialmarked.len == 1) //If there's only one of you, then don't do it. I'm not that evil.
+		return
+	chosen = pick(potentialmarked)
```

Comment:
This can pick from an empty list, make sure it does not do that.
2 runtimes on round 10605

Reply:
Fixed